### ISSUE TITLE
Lux and GraalVM ?

### ISSUE THREAD

**Issue by DjebbZ** [LuxLang/lux#44]
Hello,

I'm just a curious person that happens to use (and love) Clojure, so Lux is kinda interesting for me. With the recent announcement of Oracle's [GraalVM](https://blogs.oracle.com/developers/announcing-graalvm), did you already think of using it somehow for Lux ?

Thanks in advance for your answer.

**Comment by LuxLang** [comment]
I've been keeping my eyes on GraalVM for a while now and find it very exciting.

While there is likely no direct benefit to using Graal with Lux (since Lux already compiles to JVM bytecode, and doesn't have the dynamic features of dynamic languages that would benefit from Graal's JIT technology), there are some indirect benefits.

The people behind Graal have implemented interpreters for some languages, to get them to run on the JVM. One of them being LLVM bitcode (which is going to be very useful when adding an LLVM backend to Lux).

The current strategy for adding new backends to Lux involves using interpreters.
Graal (and its associated Truffle framework) would make it easier to create interpreters for new platforms Lux might want to target, in case nobody has written such interpreters for the JVM before.

So, in short, Graal offers no direct benefit to Lux, but plenty of indirect benefits, in terms of which platforms it can serve as a gateway to.

**Comment by DjebbZ** [comment]
Thanks for the fast and detailed answer. It makes sense.

Keep up the good work !
